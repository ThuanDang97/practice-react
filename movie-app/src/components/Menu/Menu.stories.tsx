import { Story } from '@storybook/react';
import Menu from './Menu';
import { TProps } from './Menu';

export default {
  title: 'Components/Menu',
  component: Menu,
};

const Template: Story<TProps> = (args): JSX.Element => <Menu {...args} />;

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {
  title: 'Menu',
  children: 'Children',
};
