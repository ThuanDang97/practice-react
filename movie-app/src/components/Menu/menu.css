.panel-card {
  min-width: 260px;
  width: 260px;
  border: 1px solid #e3e3e3;
  border-radius: var(--imageBorderRadius);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  overflow: hidden;
  box-shadow: 0 2px 8px rgb(0 0 0 / 10%);
  background-color: #fff;
}

.panel-card + .panel-card {
  margin-top: 30px;
}

.name {
  padding: 14px 16px;
}

.name h2 {
  font-size: 20px;
}

.filter {
  border-top: 1px solid #eee;
  padding: 14px 16px 16px 16px;
}

.filter h3 {
  margin: 0;
  padding: 0;
  display: inline-flex;
  align-items: center;
  width: 100%;
  font-size: 1em;
  font-weight: 300;
  margin-bottom: 10px;
}

.widget {
  width: 100%;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue',
    Arial, 'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol', 'Noto Color Emoji';
  line-height: 1.5;
  text-align: left;
  white-space: nowrap;
  display: flex;
  flex-direction: column;
  row-gap: 15px;
}

.dropdown {
  border-color: #e4e7eb;
  color: #212529;
  background-color: #e4e7eb;
  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out,
    border-color 0.2s ease-in-out, box-shadow 0.2s ease-in-out;
  border-radius: 0.25rem;
  padding: 0;
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
  border-width: 1px;
  border-style: solid;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  outline: 0;
}

.dropdown:hover {
  border-color: #c7cdd5;
  color: #212529;
  background-color: #ced3db;
}

.input {
  margin: 0;
  padding: 0.375rem 0.75rem;
  font-size: 14px;
  width: 100%;
  min-width: 0;
  display: flex;
  align-items: center;
  flex: 0 1 auto;
}

.select {
  padding: 10px 11px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.icon-select {
  border: solid black;
  border-width: 0 2px 2px 0;
  padding: 4px;
  transform: rotate(45deg);
}

.sub-menu {
  position: absolute;
  width: 226px;
  background: #fff;
  border: 1px solid #c7cdd5;
  border-radius: 0.25rem;
}

.sub-list {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  z-index: 1;
}

.item {
  padding: 5px 15px;
  cursor: pointer;
}

.item:hover {
  background-color: #f8f9fa;
}
