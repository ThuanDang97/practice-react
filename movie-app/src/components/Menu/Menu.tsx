import { memo } from 'react';
import './menu.css';

export type TProps = {
  title: string;
  children: React.ReactNode;
};

const Menu = (props: TProps): JSX.Element => {
  const { title, children } = props;

  return (
    <div className="panel-card">
      <div className="name">
        <h2>{title}</h2>
      </div>
      <div className="filter">
        <span className="widget" data-testid="sub-menu">
          {children}
        </span>
      </div>
    </div>
  );
};

export default memo(Menu);
