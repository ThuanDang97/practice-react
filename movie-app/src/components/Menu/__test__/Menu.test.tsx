import { render, screen } from '@testing-library/react';
import Menu from '../Menu';

describe('Test Menu component', () => {
  const component = <Menu title={'Menu'}> Sub Menu </Menu>;

  test('Component Menu matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Menu should render correctly', () => {
    render(component);

    expect(screen.getByRole('heading')).toHaveTextContent('Menu');
    expect(screen.getByTestId('sub-menu')).toHaveTextContent('Sub Menu');
  });
});
