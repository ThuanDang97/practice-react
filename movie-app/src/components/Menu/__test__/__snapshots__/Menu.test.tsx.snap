// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Menu component Component Menu matches DOM Snapshot 1`] = `
<div>
  <div
    class="panel-card"
  >
    <div
      class="name"
    >
      <h2>
        Menu
      </h2>
    </div>
    <div
      class="filter"
    >
      <span
        class="widget"
        data-testid="sub-menu"
      >
         Sub Menu 
      </span>
    </div>
  </div>
</div>
`;
